File: qqbot-gateway/src/main/java/com/mieai/qqbot/gateway/GatewaySessionState.java
package com.mieai.qqbot.gateway;

/** Observable lifecycle states of one bot Gateway session. */
public enum GatewaySessionState {
    NEW,
    LOADING_SNAPSHOT,
    CONNECTING,
    AWAITING_HELLO,
    IDENTIFYING,
    RESUMING,
    READY,
    BACKING_OFF,
    STOPPED
}

File: qqbot-gateway/src/main/kotlin/com/mieai/qqbot/gateway/GatewaySessionState.kt
package com.mieai.qqbot.gateway

/** Observable lifecycle states of one bot Gateway session. */
enum class GatewaySessionState {
    NEW,
    LOADING_SNAPSHOT,
    CONNECTING,
    AWAITING_HELLO,
    IDENTIFYING,
    RESUMING,
    READY,
    BACKING_OFF,
    STOPPED,
}
